include/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.hpp

```diff
@@ -133,18 +136,29 @@ class Propagator
 
     /// @brief              Calculate the state at an instant, given initial state
     /// @code
-    ///                     State state = propagator.calculateStateAt(aState, anInstant) ;
+    ///                     State state = propagator.calculateStateAt(aState, anInstant);
     /// @endcode
     /// @param              [in] aState An initial state
     /// @param              [in] anInstant An instant
     /// @return             State
 
     State calculateStateAt(const State& aState, const Instant& anInstant) const;
 
+    /// @brief              Calculate the state subject to an Event Condition, given initial state and maximum end time
+    /// @code
+    ///                     State state = propagator.calculateStateAt(aState, anInstant, anEventCondition);
+    /// @endcode
+    /// @param              [in] aState An initial state
+    /// @param              [in] anInstant An instant
+    /// @param              [in] anEventCondition An event condition
+    /// @return             State
+
+    State calculateStateAt(const State& aState, const Instant& anInstant, const EventCondition& anEventCondition) const;
```

Comment:
So, eventually when we want to propagate to the first of many viable conditions, we can just group them under the OR condition and pass it here right?

Just a question, resolve if what I said is correct : )

Reply:
Yes that is indeed the case. In a separate MR I will add the Conjunctive/Disjunctive classes to enable this.